### ISSUE TITLE
[new-cap] false positives about upper camel cased decorators

### ISSUE THREAD

**Issue by iwata** [eslint/typescript-eslint-parser#569]
**What version of TypeScript are you using?**
`3.1.6`

**What version of `typescript-eslint-parser` are you using?**
`21.0.1`

**What code were you trying to parse?**

```ts
import { Vue, Component } from 'vue-property-decorator'
import Hello from './hello.vue'

@Component({
  components: {Hello}
})
export default class Header extends Vue {
   private title = ''
}
```

**What did you expect to happen?**
All is good.

**What happened?**
Error that `A function with a name starting with an uppercase letter should only be used as a constructor.` on `@Component({...})`.

**Comment by kaicataldo** [comment]
I don't think we'll be able to fix this in ESLint core by default, but I believe you can use [capIsNewExceptions](https://eslint.org/docs/rules/new-cap#capisnewexceptions) to work around this.

**Comment by iwata** [comment]
@kaicataldo 
Thanks!
I'm working good.
